### PR TITLE
Better Error Handling

### PR DESCRIPTION
**Pull request by malnick**
@darkonie If you have a minute please review, thanks.

### CODE REVIEW

File: signal/diagnostics.go [hunk 1]
```diff
@@ -82,12 +82,12 @@ func (d *Diagnostics) getMethod() string {
 	return d.Method
 }
 
-func (d *Diagnostics) getError() string {
+func (d *Diagnostics) getError() []string {
 	return d.Error
 }
 
 func (d *Diagnostics) setError(err string) {
```
Comment (darkonie) [inline signal/diagnostics.go]: maybe `addError` or `appendError` ?

File: signal/mesos.go [hunk 2]
```diff
@@ -73,12 +73,12 @@ func (d *Mesos) getMethod() string {
 	return d.Method
 }
 
-func (d *Mesos) getError() string {
+func (d *Mesos) getError() []string {
 	return d.Error
 }
 
 func (d *Mesos) setError(err string) {
```
Comment (darkonie) [inline signal/mesos.go]: same as above

File: signal/reporter.go [hunk 3]
```diff
@@ -39,71 +39,68 @@ type Reporter interface {
...
+		// ex. headerName = "Content-Type" and headerValue = "application/json"
+		req.Header.Add(headerName, headerValue)
+	}
+	log.Debugf("Request %s: %+v", endpoint, req)
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == 200 {
```
Comment (darkonie) [inline signal/reporter.go]: if resp.StatusCode != 200 {
   return errors.New(...

...
READ BODY HERE
return nil

File: signal/signal.go [hunk 4]
```diff
@@ -20,21 +20,27 @@ var (
...
-			done <- r
-			return err
-		}
-
-		err = r.setTrack(c)
-		if err != nil {
-			r.setError(err.Error())
-			done <- r
-			return err
+		if len(r.getEndpoints()) != 0 {
```
Comment (darkonie) [inline signal/signal.go]: if len(r.getEndpoints()) == 0 {
    return error
}
<REST OF THE CODE>

File: signal/signal.go [hunk 5]
```diff
@@ -85,10 +91,12 @@ func executeRunner(c config.Config) error {
 		select {
 		case r := <-done:
 			if len(c.TestURL) > 0 {
-				log.Info("Adding test data")
+				log.Debugf("Adding test data for %s: %+v", r.getName(), r.getTrack())
 				tester[r.getName()] = r.getTrack()
 			} else if len(r.getError()) > 0 {
-				log.Errorf("%s: %s", r.getName(), r.getError())
+				for _, err := range r.getError() {
+					log.Errorf("%s: %s", r.getName(), err)
```
Comment (darkonie) [inline signal/signal.go]: i am not quite sure what benefits it gives you to collect the errors and then log them all.
Maybe it is better to log the error right away? Might be more useful when debugging

**Comment by malnick** [comment]
Sorry, accidentally merged this without reloading the page first and missed all these comments. Will submit a PR to back these in.

**Pull request merged**